client/src/features/Catalogue/components/CatalogMainFilter/CatalogMainFilter.js

```diff
@@ -57,24 +68,75 @@ const ResetButton = styled((props) => (
   },
 }));
 
-const CatalogMainFilter = () => (
-  <FilterWrapper>
-    <Typography variant="h3">Filter</Typography>
+const CatalogMainFilter = () => {
+  const dispatch = useDispatch();
+  const prices = useSelector((store) => store.filter.prices);
+  const [minPrice, maxPrice] = prices;
+  const products = useSelector((state) => state.catalogue.products, shallowEqual);
+  const allPrices = [...new Set(products.map((tour) => tour.currentPrice))].sort((a, b) => a - b);
+  const duration = useSelector((store) => store.filter.duration);
+  const seasons = useSelector((store) => store.filter.seasons.map((el) => (el ? el.toLowerCase() : el)));
 
-    <Grid container columnSpacing={5} sx={{ p: 0 }}>
-      <Grid spacing={4} item xs={12} tablet={6} laptop={12}>
-        <CatalogFilterPrice />
-        <CatalogFilterTourists />
-      </Grid>
+  useEffect(() => {
+    dispatch(getProducts());
+  }, []);
+
+  const filterTours = async () => {
+    const filterPrices = () => {
+      if (minPrice > maxPrice) {
+        const reversePrices = [...prices].reverse();
+        dispatch(setPrices(reversePrices));
+        const filterPricesValues = allPrices.filter((price) => price >= maxPrice && price <= minPrice);
+        return filterPricesValues;
+      }
+      const filterPricesValues = allPrices.filter((price) => price >= minPrice && price <= maxPrice);
+      return filterPricesValues;
+    };
+
+    const params = new URLSearchParams();
+
+    params.set('currentPrice', filterPrices());
+    if (duration.length > 0) {
+      params.set('duration', duration);
+    }
+    if (seasons.length > 0) {
+      params.set('season', seasons.concat('all seasons'));
+    }
+
+    dispatch(setIsLoading(true));
+    dispatch(setIsFilter(true));
+    await dispatch(fetchFilteredTours(params));
+    dispatch(setIsLoading(false));
+  };
+
+  const resetFilter = () => {
+    dispatch(setIsFilter(false));
+    dispatch(setFilteredTours([]));
+    dispatch(setPrices([Math.min.apply(null, allPrices), Math.max.apply(null, allPrices)]));
+    dispatch(setClearDuration([]));
+    dispatch(setClearDuration([]));
+    dispatch(setAllSeasons([]));
+  };
+
```

Comment:
You can create another reducer in your slice that will changed your current state on initial 
you can write me if you have any questions about it